packages/jest-cli/src/test_scheduler.js

```diff
@@ -96,6 +96,23 @@ export default class TestScheduler {
         });
         return Promise.resolve();
       }
+      if (testResult.leaks) {
+        const message = [
+          'Your test suite is leaking memory! Please ensure all references are cleaned.',
+          '',
+          'There is a number of things that can leak memory:',
+          '  - Async operations that have not finished (e.g. fs.readFile).',
+          '  - Timers not properly mocked (e.g. setInterval, setTimeout).',
+          '  - Missed global listeners (e.g. process.on).',
+          '  - Keeping references to the global scope.',
+        ].join('\n');
+
+        await onFailure(test, {
+          message,
+          stack: new Error(message).stack,
+        });
+        return Promise.resolve();
```

Comment:
`return` is enough, no? It's already an async function.

Reply:
All this `Promise.resolve()` (not only mine, but all others) are there because of the `eslint` `consistent-return` rule 😞, but TBH I prefer keeping them and not adding an `eslint-disable`.

Reply:
it's sufficient to just `return onFailure(...)` since it's already a promise (same thing can be fixed in `testResult.testResults.length` block above)